examples/chip-tool/commands/common/Commands.cpp

```diff
@@ -41,6 +42,35 @@ constexpr const char * kJsonCommandKey              = "command";
 constexpr const char * kJsonCommandSpecifierKey     = "command_specifier";
 constexpr const char * kJsonArgumentsKey            = "arguments";
 
+template <typename T>
+struct HasInitWithString
+{
+    template <typename U>
+    static constexpr auto check(U *) -> typename std::is_same<decltype(std::declval<U>().Init("")), CHIP_ERROR>::type;
+
+    template <typename>
+    static constexpr std::false_type check(...);
+
+    typedef decltype(check<std::remove_reference_t<T>>(nullptr)) type;
+
+public:
+    static constexpr bool value = type::value;
+};
+
+// Template so we can do conditional enabling
+template <typename T, std::enable_if_t<HasInitWithString<T>::value, int> = 0>
+static void UseStorageDirectory(T & storageManagerImpl, const char * storageDirectory)
+{
+#if !CHIP_DISABLE_PLATFORM_KVS
+    std::string platformKVS = std::string(storageDirectory) + "/chip_tool_kvs";
```

Comment:
Why not allow giving a filename on command-line like `--KVS` in all the examples? This would be better than forcing `chip_tool_kvs` name.

Reply:
Because we already have a way of specifying the storage dir as a whole on the command line and store a bunch of different files there.  We _could_ also have a way to configure the name of this one file inside the already-configurable dir, but in practice people who want to control this will just set the dir.